astropy/cosmology/connect.py

```diff
@@ -237,8 +246,8 @@ class CosmologyToFormat(io_registry.UnifiedReadWrite):
         Keyword arguments passed through to data writer.
     """
 
-    def __init__(self, instance, cls):
+    def __init__(self, instance: Cosmology, cls: type[Cosmology]) -> None:
         super().__init__(instance, cls, "write", registry=convert_registry)
 
-    def __call__(self, format, *args, **kwargs):
+    def __call__(self, format: str, *args: Any, **kwargs: Any) -> Any:
```

Comment:
...but this wrapper returns `Any`?

Reply:
It's not Liskov Substitution safe. That's for sure.

Reply:
I suppose in that case the best thing to do would be to change the type of `format` to be `Literal[...]` and to [overload](https://docs.python.org/3/library/typing.html#typing.overload) the method for the different available formats.

Reply:
We could do this for the built-in formats. But in general this is a dynamic registry that users can hook into, so the base of the overloads needs to remain `(:str, ...) -> :Any`.

Reply:
It looks like we do need a base case with `-> Any`, but implementing overloads would at least be useful for the built-in formats.

Reply:
Done!